workspaces.tf

```
@@ -1,18 +1,26 @@
 locals {
-  deployment_name = var.deployment_name
+  workspaces = var.workspaces
+  add_deployment_name = var.add_deployment_name
 }
 
-resource "databricks_mws_workspaces" "this" {
+resource "databricks_mws_workspaces" "these" {
+  for_each        = toset(local.workspaces)
+  
   account_id      = var.databricks_account_id
   aws_region      = var.aws_region
-  workspace_name  = var.resource_prefix
-  deployment_name = local.deployment_name
+  workspace_name  = each.key
+  deployment_name = local.add_deployment_name ? each.key : null
 
-  credentials_id           = databricks_mws_credentials.this.credentials_id
-  storage_configuration_id = databricks_mws_storage_configurations.this.storage_configuration_id
-  network_id               = databricks_mws_networks.this.network_id
+  credentials_id           = databricks_mws_credentials.these.credentials_id
+  storage_configuration_id = databricks_mws_storage_configurations.these.storage_configuration_id
+  network_id               = databricks_mws_networks.these.network_id
```

:warning: watch out, those resources haven't been renamed as they are a single one!
```suggestion
  credentials_id           = databricks_mws_credentials.this.credentials_id
  storage_configuration_id = databricks_mws_storage_configurations.this.storage_configuration_id
  network_id               = databricks_mws_networks.this.network_id
```

---

good catch man, thanks